apps/address-service/domains/common/utils/services/suggest/SuggestionKeystoneApp.js

```diff
@@ -0,0 +1,76 @@
+const express = require('express')
+
+const get = require('lodash/get')
+const { SuggestionProviderDetector } = require('@address-service/domains/common/utils/services/suggest/SuggestionProviderDetector')
+
+class SuggestionKeystoneApp {
+    /**
+     * @param params
+     * @returns {Express}
+     */
+    prepareMiddleware (params) {
+        const app = express()
+        const providerDetector = new SuggestionProviderDetector()
+
+        function setNoCache (req, res, next) {
+            res.set('Pragma', 'no-cache')
+            res.set('Cache-Control', 'no-cache, no-store')
+            next()
+        }
+
+        app.get('/suggest', setNoCache, async (req, res, next) => {
+            /**
+             * Using to detect a proper suggestion provider
+             * @type {string}
+             */
+            const geo = String(get(req, ['query', 'geo'], null))
+
+            /**
+             * User's search string
+             * @type {?string}
+             */
+            const s = get(req, ['query', 's'])
+
+            /**
+             * To skip normalization of results. If true - returns raw provider's result
+             * @type {boolean}
+             */
+            const bypass = Boolean(get(req, ['query', 'bypass'], false))
+
+            /**
+             * We need different query parameters for server side and for client side
+             * At least for dadata
```

Comment:
is it ok to put some provider-specific args here? may be we need to remove it?

Reply:
All this provider/client stuff was moved to contexts.